File: app/src/main/java/me/imzack/app/ender/injector/component/TypeEditComponent.java
package me.imzack.app.ender.injector.component;

import me.imzack.app.ender.injector.module.TypeEditPresenterModule;
import me.imzack.app.ender.injector.scope.ActivityScope;
import me.imzack.app.ender.view.activity.TypeEditActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = TypeEditPresenterModule.class)
public interface TypeEditComponent {

    void inject(TypeEditActivity typeEditActivity);
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/TypeEditComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.TypeEditPresenterModule
import net.zackzhang.app.end.injector.scope.ActivityScope
import net.zackzhang.app.end.view.activity.TypeEditActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(TypeEditPresenterModule::class))
interface TypeEditComponent {

    fun inject(typeEditActivity: TypeEditActivity)
}
